File: app/src/main/java/com/wuruoye/know/model/beans/RecordType.java
package com.wuruoye.know.model.beans;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created at 2019/3/18 12:31 by wuruoye
 * Description:
 */
public class RecordType implements Parcelable {
    private int id;
    private String title;
    private List<RecordView> views;
    private long createTime;
    private long updateTime;

    public RecordType(int id, String title, List<RecordView> views,
                      long createTime, long updateTime) {
        this.id = id;
        this.title = title;
        this.views = views;
        this.createTime = createTime;
        this.updateTime = updateTime;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<RecordView> getViews() {
        return views;
    }

    public void setViews(List<RecordView> views) {
        this.views = views;
    }

    public long getCreateTime() {
        return createTime;
    }

    public void setCreateTime(long createTime) {
        this.createTime = createTime;
    }

    public long getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(long updateTime) {
        this.updateTime = updateTime;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.id);
        dest.writeString(this.title);
        dest.writeList(this.views);
        dest.writeLong(this.createTime);
        dest.writeLong(this.updateTime);
    }

    public RecordType() {
    }

    protected RecordType(Parcel in) {
        this.id = in.readInt();
        this.title = in.readString();
        this.views = new ArrayList<RecordView>();
        in.readList(this.views, RecordView.class.getClassLoader());
        this.createTime = in.readLong();
        this.updateTime = in.readLong();
    }

    public static final Creator<RecordType> CREATOR = new Creator<RecordType>() {
        @Override
        public RecordType createFromParcel(Parcel source) {
            return new RecordType(source);
        }

        @Override
        public RecordType[] newArray(int size) {
            return new RecordType[size];
        }
    };
}

File: app/src/main/java/com/wuruoye/know/model/beans/RecordType.kt
package com.wuruoye.know.model.beans

import android.os.Parcel
import android.os.Parcelable
import com.wuruoye.know.util.sql.table.RecordTypeTable

/**
 * Created at 2019/3/18 12:31 by wuruoye
 * Description:
 */
class RecordType(var id: Int = -1,
                 var title: String = "",
                 var views: ArrayList<RecordView> = arrayListOf(),
                 var createTime: Long = -1,
                 var updateTime: Long = -1
) : Parcelable {
    constructor(table: RecordTypeTable, views: ArrayList<RecordView>): this(table.id,
            table.title, views, table.createTime, table.updateTime)

    constructor(title: String): this() {
        this.title = title
    }

    constructor(source: Parcel) : this(
            source.readInt(),
            source.readString(),
            ArrayList<RecordView>().apply { source.readList(this, RecordView::class.java.classLoader) },
            source.readLong(),
            source.readLong()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeInt(id)
        writeString(title)
        writeList(views)
        writeLong(createTime)
        writeLong(updateTime)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<RecordType> = object : Parcelable.Creator<RecordType> {
            override fun createFromParcel(source: Parcel): RecordType = RecordType(source)
            override fun newArray(size: Int): Array<RecordType?> = arrayOfNulls(size)
        }
    }
}
